demos/python_demos/object_detection_demo_yolov3_async/object_detection_demo_yolov3_async.py

```diff
@@ -219,154 +321,172 @@ def main():
+                                                                                  obj['xmin'], obj['ymin'], obj['xmax'],
+                                                                                  obj['ymax'],
+                                                                                  color))
+
+                cv2.rectangle(frame, (obj['xmin'], obj['ymin']), (obj['xmax'], obj['ymax']), color, 2)
+                cv2.putText(frame,
+                            "#" + det_label + ' ' + str(round(obj['confidence'] * 100, 1)) + ' %',
+                            (obj['xmin'], obj['ymin'] - 7), cv2.FONT_HERSHEY_COMPLEX, 0.6, color, 1)
+
+            # Draw performance stats over frame
+            if mode_info[mode.current].frames_count != 0:
+                fps_message = "FPS: {:.1f}".format(mode_info[mode.current].frames_count / \
+                                                   (perf_counter() - mode_info[mode.current].last_start_time))
+                mode_info[mode.current].latency_sum += perf_counter() - start_time
+                latency_message = "Latency: {:.1f} ms".format((mode_info[mode.current].latency_sum / \
+                                                              mode_info[mode.current].frames_count) * 1e3)
+
+                put_highlighted_text(frame, fps_message, (15, 20), cv2.FONT_HERSHEY_COMPLEX, 0.75, (200, 10, 10), 2)
+                put_highlighted_text(frame, latency_message, (15, 50), cv2.FONT_HERSHEY_COMPLEX, 0.75, (200, 10, 10), 2)
+            
+            mode_message = "{} mode".format(mode.current.name)
+            put_highlighted_text(frame, mode_message, (10, int(origin_im_size[0] - 20)),
+                                 cv2.FONT_HERSHEY_COMPLEX, 0.75, (10, 10, 200), 2)
+
+            if not args.no_show:
+                cv2.imshow("Detection Results", frame)
+                key = cv2.waitKey(wait_key_time)
+
+                if key in {ord("q"), ord("Q"), 27}: # ESC key
+                    break
+                if key == 9: # Tab key
+                    prev_mode = mode.current
+                    mode.next()
+                    
+                    await_requests_completion(exec_nets[prev_mode].requests)
+                    for active_request in active_requests_data.values():
+                        add_completion_result(active_request)
+                    active_requests_data.clear()
+                    
+                    mode_info[prev_mode].last_end_time = perf_counter()
+                    mode_info[mode.current] = ModeInfo()
+                else:
+                    presenter.handleKey(key)
+
+        elif cap.isOpened():
+            start_time = perf_counter()
             ret, frame = cap.read()
+            if not ret:
+                if args.loop_input:
+                    if number_input_frames != 1:
+                        cap.set(cv2.CAP_PROP_POS_FRAMES, 0)
+                    else:
+                        cap.open(input_stream)
```

Comment:
Since `CAP_PROP_POS_FRAMES` doesn't work for an image, I think it is simpler just to reopen `cap` independently on `number_input_frames`